test/functional/feature_discover.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -discover command."""
+
+import socket
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+def is_valid_ipv4_address(address):
+    try:
+        socket.inet_aton(address)
+    except socket.error:
+        return False
+    return True
+
+
+def is_valid_ipv6_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET6, address)
+    except socket.error:
+        return False
+    return True
+
+
+class DiscoverTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def validate_addresses(self, addresses_obj):
+        for address_obj in addresses_obj:
+            address = address_obj['address']
+            self.log.info(f"Validating {address}")
+            valid = (is_valid_ipv4_address(address)
+                     or is_valid_ipv6_address(address))
+            assert_equal(valid, True)
+
+    def test_local_addresses(self, test_case, should_be_empty=False):
```

Comment:
nit, maybe use named args (for all, or after the first one), maybe better to omit a default value if you're passing values in for clarity, and naming suggestion 
```diff
-    def test_local_addresses(self, test_case, should_be_empty=False):
+    def test_local_addresses(self, test_case, *, expect_empty=False):

...

         for test_case in test_cases:
-            self.test_local_addresses(test_case, False)
+            self.test_local_addresses(test_case, expect_empty=False)
 
         for test_case in test_cases_empty:
-            self.test_local_addresses(test_case, True)
+            self.test_local_addresses(test_case, expect_empty=True)
```

Reply:
Great, done!